src/sdk/wire_bytes.rs

```diff
@@ -27,41 +28,62 @@ pub fn encode<T: Serialize>(payload: &T) -> TofnResult<BytesVec> {
     })
 }
 
+/// Serialize a value using bincode and log errors
 pub fn serialize<T: ?Sized>(value: &T) -> TofnResult<BytesVec>
 where
     T: serde::Serialize,
 {
-    match bincode::serialize(value) {
-        Ok(bytes) => Ok(bytes),
-        Err(err) => {
-            error!("serialization failure: {}", err.to_string());
-            Err(TofnFatal)
-        }
-    }
+    // Create serialization options for bincode.
+    // The default options don't bound pre-allocation size,
+    // use little-endian and varint encoding, and reject trailing bytes.
+    let options = DefaultOptions::new()
+        .with_no_limit()
+        .with_big_endian()
+        .with_varint_encoding()
```

Comment:
varint encoding saves a lot in smaller messages

Reply:
```suggestion
        .with_varint_encoding() // saves a lot of space in smaller messages
```